lib/resty/rediscluster.lua

```diff
@@ -551,17 +551,19 @@ function _M.init_slots(self)
         return true
     end
 
-    local lock, elapsed, err
-    lock, err = resty_lock:new(self.config.dict_name or DEFAULT_SHARED_DICT_NAME)
+    local lock, elapsed, err, err_msg
+    lock, err = resty_lock:new(self.config.dict_name or DEFAULT_SHARED_DICT_NAME, { timeout = self.config.lock_timeout })
     if not lock then
-        ngx_log(NGX_ERR, "failed to create lock in initializing slot cache: ", err)
-        return nil, err
+        err_msg = "failed to create lock in initializing slot cache: " .. tostring(err)
```

Comment:
Should we make the same change in `refresh_slots`?

Reply:
Currently, this function's return values are not checked. The log in only printed as a hint. Error in this function is relatively trivial.